pkg/vmsock/conn_linux.go

```diff
@@ -88,13 +88,19 @@ func handleUDP(uConn *net.UDPConn, addr *net.UDPAddr, b []byte) {
 		return
 	}
 	data := make([]byte, UDPMaxBufer)
-	_, err = conn.Read(data)
+	n, err := conn.Read(data)
 	if err != nil {
 		log.Errorf("handleUDP read from vsock host: %v", err)
 		return
 	}
-	// remove the tcp length from the beginning since that is the place for msg id ;b
-	_, err = uConn.WriteToUDP(data[2:], addr)
+	// data[2:n] is to remove the tcp length from the beginning
```

Comment:
I find the wording a bit unclear; maybe have a temporary variable to attach the comment to?
```go
	// TCP DNS requests have two-byte length prefix, which is not present in UDP
	// DNS requests (See RFC 1035).  Drop the prefix; however, note that the
	// length of the write needs to match the length of the received slice (as
	// opposed to its capacity).  This is necessary since some tools such as
	// busybox nslookup explicitly check this.
	reqBody := data[2:n]
	_, err = uConn.WriteToUDP(reqBody, addr)
```

Reply:
Actually, given that the first two bytes of `data` is a length, we should use that instead of the return value of `Read`.  This can matter if we have an over-read.

Reply:
> This can matter if we have an over-read.

@mook-as could you elaborate? I'm trying to understand how we can have an over-read in UDP segments?

Reply:
You're reading vsock, not UDP (but _writing_ to UDP), right?

Reply:
We are reading `[]byte` that coming out of vsock and the slice of byte represent a UDP datagram, I believe the vsock is just a form of transport or connection to the other end that talks UDP. The connection itself is the same object that is shared with `net.Conn` or `net.UDPConn` which is independent of the bytes themselves.  Hence, the original caller such as nslookup, dig, or any other DNS client knows nothing about vsock.

Reply:
Discussed with Nino: the truncation stuff is going to have to be done via two AF_VSOCK connections to the host, as we should only truncate on UDP, but we don't know if a request is UDP (and there's no good way for us to figure that out by the time we should truncate).

This will need to be done in a separate PR.